hpack/js/src/main/scala/org/http4s/hpack/HpackUtilPlatform.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2022 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright 2014 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.http4s.hpack
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSImport
+import scala.scalajs.js.typedarray.Int8Array
+import scala.annotation.nowarn
+
+private[hpack] class HpackUtilPlatform {
+
+  // the property holding the Uint8Array backing the Array[Byte] implementation
+  // this lets us avoid having to copy it via the provided APIs
+  private[this] val array0Property =
+    js.Object.getOwnPropertyNames((new Array[Byte](0)).asInstanceOf[js.Object])(0)
```

Comment:
This sort of hackery and reliance on internal implementation details makes me nervous, but we've used this same technique successfully in CE3 to enable tracing on JS. I've also setup CI to test both in both fast and full optimization mode.